### ISSUE TITLE
question about paper

### ISSUE THREAD

**Issue by ChdDongyang** [researchmm/MM-Diffusion#9]
May I ask what audio encoders, video encoders, and corresponding decoders are used in the model? It seems that the paper did not mention.

**Comment by SoloMannn** [comment]
Hi! I found the encoder in the code and I will paste for you. It`s just a convlution layer.
`class InitialBlock(nn.Module):
    def __init__(
        self,
        video_in_channels,
        audio_in_channels,
        video_out_channels,
        audio_out_channels,
        kernel_size = 3 
    ):
        super().__init__()
        self.video_conv = VideoConv(video_in_channels, video_out_channels, kernel_size, conv_type='2d+1d')
        self.audio_conv = AudioConv(audio_in_channels, audio_out_channels, kernel_size, conv_type='linear')

    def forward(self, video, audio): 
        return self.video_conv(video), self.audio_conv(audio)

`
`class VideoConv(nn.Module):
    def __init__(
        self,
        in_channels,
        out_channels,
        kernel_size=3,
        stride = 1,
        padding = "same",
        dilation = 1,
        conv_type = '2d+1d'
    ):
        super().__init__()
        self.conv_type = conv_type
        self.padding= padding

        if conv_type == '2d+1d':
            self.video_conv_spatial = conv_nd(2, in_channels, out_channels, kernel_size, stride, padding, dilation)
            self.video_conv_temporal = conv_nd(1, out_channels,out_channels, kernel_size, stride, padding, dilation)
        elif conv_type == '3d':
            self.video_conv = conv_nd(3, in_channels, out_channels,kernel_size, stride=stride, padding=padding, dilation=dilation)
        else:
            raise NotImplementedError`

**Comment by ChdDongyang** [comment]
@SoloMannn OK, thanks!!!

**Comment by henu77** [comment]
The reply disappeared, can you tell me the answer? @ChdDongyang
